CRM/Core/DAO.php

```diff
@@ -2782,6 +2782,16 @@ public static function createSQLFilter($fieldName, $filter, $type = NULL, $alias
       if (in_array($operator, self::acceptedSQLOperators(), TRUE)) {
         switch ($operator) {
           // unary operators
+          case 'IS EMPTY':
+            if ($type === 'String') {
+              return (sprintf('%s', "$fieldName = '' OR $fieldName IS NULL"));
+            }
+
+          case 'IS NOT EMPTY':
+            if ($type === 'String') {
+              return (sprintf('%s', "$fieldName <> '' AND $fieldName IS NOT NULL"));
+            }
```

Comment:
I'd like to move this code into apiv4. In fact I'd like to copy this whole function into APIv4.

Reply:
But aside from code placement, I have a question about functionality. Namely, what happens if the type is not string?

Reply:
Hang on, reading the code, `$type` cannot be anything but NULL because that param is not used.

Reply:
But before I do my own version of this PR I could use some help understanding what's going on here @eileenmcnaughton

Reply:
@colemanw $type IS passed in because of the change to API select query in this PR- so only apiv4 would pass it in.

If the function type is not a string then it reverts back to IS NULL - there might be some other string-like parameters we should treat like strings but for something like a BOOL 'NO' is not empty whereas NULL is